### PR TITLE
Fix metadata

### LINKED ISSUE #13: Thoughts about a metadata object
Currently we create, harmonize and store metadata in geojson for each source directly in [construct_metadata ](https://github.com/dlr-eoc/ukis-pysat/blob/master/ukis_pysat/download.py#L161) method. This works fine at the moment, but is errorprone in case we may want to expand the metadata attributes, add further sources or modify existing metadata at some stage. Therefore, it would be good to have a dedicated metadata object with clearly defined attributes, dtypes, constraints, validation routines and descriptions.

An interesting approach to solve this in a bit more flexible way than just using a dictionary could be [dataclasses ](https://docs.python.org/3/library/dataclasses.html). They have been added to the Python standard library since 3.7. This would restrict us to use only Python>=3.7. 

What are your thoughts on such a restriction and the possible gains from a dataclass?

### PR DESCRIPTION
**Pull request by MWieland**
Closes #13 Adds dependency to pyfields and pandas. Pandas is not critical but a nice package to inspect metadata. Pandas is anyway already a dependency of sentinelsat.

### CODE REVIEW

**Comment by fwfichtner** [comment]
@MWieland it failed because it was running a merged version.

File: ukis_pysat/download.py [hunk 1]
```diff
@@ -382,6 +346,156 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+
+
+class MetadataCollection:
+    """
+    Provides a container to store a collection of metadata objects. Conversion methods are provided to
+    analyse the metadata collection further.
+    """
+
+    __init__ = make_init()
+    items: List[Metadata] = field(type_hint=List[Metadata])
```
Comment (fwfichtner) [inline ukis_pysat/download.py:444]: Shouldn't this type alias be alist already? I don't really get this, but could also be me.

File: ukis_pysat/download.py [hunk 2]
```diff
@@ -382,6 +346,156 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+        """
+        return [item.to_geojson() for item in self.items]
+
+    def to_pandas(self):
+        """Converts MetadataCollection to Pandas Dataframe.
+
+        :returns: MetadataCollection (Pandas Dataframe)
+        """
+        try:
+            import pandas as pd
```
Comment (fwfichtner) [inline ukis_pysat/download.py:473]: I hope that's okay, I thought for these convenience functions we don't need to import such a heavy library. Did the same for Dask.

File: ukis_pysat/download.py [hunk 3]
```diff
@@ -148,117 +127,78 @@ def query_metadata(self, platform, date, aoi, cloud_cover=None):
...
 
         try:
-            # construct list of harmonized metadata
+            # construct MetadataCollection from list of Metadata objects
             meta = []
             for m in meta_src:
                 meta.append(self.construct_metadata(meta_src=m))
-            return meta
+            return MetadataCollection(meta)
         except Exception as e:
```
Comment (fwfichtner) [inline ukis_pysat/download.py:135]: We have many of these general try, except blocks. What was the purpose of them again? Wouldn't it be better to just get rid of them?

File: ukis_pysat/download.py (hunk 2 above)
Comment (MWieland) [reply ukis_pysat/download.py:473]: I agree and that's a good way to solve it.

File: ukis_pysat/download.py (hunk 1 above)
Comment (MWieland) [reply ukis_pysat/download.py:444]: Need to check this one again. I think you are right, now that I look at it again.

File: ukis_pysat/download.py (hunk 3 above)
Comment (MWieland) [reply ukis_pysat/download.py:135]: Good point. Now is a good time to revise them again and remove what is not strictly needed.

File: ukis_pysat/download.py (hunk 1 above)
Comment (fwfichtner) [reply ukis_pysat/download.py:380]: We'll keep it for the merge and will check what to do with this later on.

**Pull request merged**